src/app/modals/shared/password/password.component.ts

```diff
@@ -89,17 +90,63 @@ export class PasswordComponent {
     _Actually_ unlock the wallet!
   */
 
-  rpc_unlock() {
-    this.log.i('rpc_unlock: calling unlock!');
-    this._rpc.call(this, 'walletpassphrase', [this.password, 99999, this.stakeOnly], this.rpc_unlock_success);
+  private rpc_unlock() {
+    this.log.i('rpc_unlock: calling unlock! timeout=' + this.unlockTimeout);
+    this.checkAndFallbackToStaking();
+    this._rpc.call(this, 'walletpassphrase', this.rpc_getUnlockParams(),
+      this.rpc_unlock_success,
+      this.rpc_unlock_failed
+    );
   }
 
-  rpc_unlock_success(json: Object) {
-    this.log.i('rpc_unlock_success: unlock was succesful :)');
+  private rpc_getUnlockParams() {
+    return [
+      this.password,
+      (this.stakeOnly ? 0 : this.unlockTimeout),
+      this.stakeOnly
+    ];
+  }
+
+  private rpc_unlock_success(json: Object) {
+    this.log.i('rpc_unlock_success: unlock was succesful!');
     this._rpc.call(this, 'getwalletinfo', null, this.rpc_alertEncryptionStatus);
   }
 
-  rpc_alertEncryptionStatus(json: Object) {
+  private rpc_unlock_failed(json: Object) {
+    this.log.i('rpc_unlock_failed: unlock failed - wrong password?');
+    alert('Unlock failed - password was incorrect.');
+  }
+
+  /*
+    If we're unlocking the wallet for a period of this.unlockTimeout, then check if it was staking
+    if(staking === true) then fallback to staking instead of locked after timeout!
+    else lock wallet
+  */
+  private checkAndFallbackToStaking() {
+    if (this._encryptionStatusService.getEncryptionStatusState() === 'Unlocked, staking only') {
+      const password = this.password;
+      const timeout = this.unlockTimeout;
+      const that = this;
+
+      /*
+        After unlockTimeout, unlock wallet for staking again.
+      */
+      setTimeout(
+        function() {
```

Comment:
Sublime has a function that creates setTimeout functions, I've been using but I've seen the bind() method in the rpc service. I'll start using that instead :+1: